taliforth.asm

```diff
@@ -102,6 +102,25 @@ cmpl_a:
 _done:
                 rts
 
+cmpl_inline:
```

Comment:
Tali already has a heavy-duty inline compiler (can inline compile any size routine) in native_words.asm (xt_compile_comma).  It might make sense to use that instead of this routine (or pull it out of that word).  

If you want to use that routine where it currently lives, you will need a bit of setup.  In particular, that full routine tries to look up the NT and will compile as a JSR if there isn't one.  You could:

A. Add your routine you want to compile to the dictionary with the AN (Always Native) flag set.  Then you'd just need to place its XT on the stack and call xt_compile_comma.

B. Alternatively, you can just sneak into the _compile_as_code: section of the routine.  I believe you'll need to remove the _ in the label name to make it a globally available label.  This section expects the following:
Beginning address of source code on return stack
Length on data stack
status byte in tmp3 (load with 0 to prevent underflow stripping)

Reply:
The _specials_done label might be an even better target.  This needs the following on the stack:
source_address cp length
It calls `move`, and then updates cp.

Reply:
ok i'll take a look - basically I just saw this code block repeated in three places for compilation so reflexively made a subroutine :)

Reply:
That totally makes sense.  The version we already have requires a bit more setup and will run slower, but this is during compiling.  I'm not as concerned with the speed during compiling if it will be faster and/or smaller when running.